### ISSUE TITLE
Error Get-PASAccount

### ISSUE THREAD

**Issue by akdemiryakup** [pspete/psPAS#90]
## Your Environment

* PowerShell Version:5.1
* psPAS Branch & Version:2.2
* CyberArk Version:10.4

## Expected Behaviour
I want list all users on cyberark 

## Current Behaviour
$token | Get-PASAccount -search 'test'

Test is the cyberark username

PowerShell show the following message error:Invoke-PASRestMethod : [500] Error mapping types.
Mapping types:
IEnumerable`1 -> List`1
System.Collections.Generic.IEnumerable`1[[CyberArk.PasswordVault.API.Objects.PVPassword, CyberArk.PasswordVault.API, Version=8.0.0.0, Culture=neutral, 
PublicKeyToken=40be1dbc8718670f]] -> System.Collections.Generic.List`1[[CyberArk.PasswordVault.PASWebServices.Models.AccountModel, 
CyberArk.PasswordVault.PASWebServices, Version=8.0.0.0, Culture=neutral, PublicKeyToken=40be1dbc8718670f]]
At C:\Windows\system32\WindowsPowerShell\v1.0\Modules\psPAS\Functions\Accounts\Get-PASAccount.ps1:278 char:13
+ ...   $result = Invoke-PASRestMethod -Uri $URI -Method GET -Headers $sess ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : CAWS00001E,Invoke-PASRestMethod

I want to list all users on cyberark

**Comment by pspete** [comment]
Hi @akdemiryakup  - sorry for delay in responding.

I cannot reproduce this issue. Using the same powershell, module & CyberArk versions as you, I get results returned as expected:

![image](https://user-images.githubusercontent.com/10600090/45452798-e1105280-b6d6-11e8-945b-a4f734bf840f.png)

Have you been able to move past this issue already?

**Comment by pspete** [comment]
@akdemiryakup - please comment this if still an issue for you.

**Comment by mike4campbell** [comment]
@akdemiryakup Did you resolve this issue on your own? I'm facing the same issue with the below configuration:


    PowerShell Version:4.0
    psPAS Branch & Version:2.2.10
    CyberArk Version:10.4

The query works for me when specifying the search parameter but the below code, which should return all accounts in the vault, produce the error.


$token | Get-PASAccount


Invoke-PASRestMethod : [500] Error mapping types.
Mapping types:
IEnumerable`1 -> List`1
System.Collections.Generic.IEnumerable`1[[CyberArk.PasswordVault.API.Objects.PVPassword, CyberArk.PasswordVault.API, 
Version=8.0.0.0, Culture=neutral, PublicKeyToken=40be1dbc8718670f]] -> 
System.Collections.Generic.List`1[[CyberArk.PasswordVault.PASWebServices.Models.AccountModel, 
CyberArk.PasswordVault.PASWebServices, Version=8.0.0.0, Culture=neutral, PublicKeyToken=40be1dbc8718670f]]
At C:\Program Files\WindowsPowerShell\Modules\psPAS\psPAS\Functions\Accounts\Get-PASAccount.ps1:278 char:13
+         $result = Invoke-PASRestMethod -Uri $URI -Method GET -Headers $sessionToken -W ...
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : CAWS00001E,Invoke-PASRestMethod

**Comment by SphericalMishap** [comment]
I'm also seeing the same error even when recreating the `Get-PASAccount -filter "Safename eq TargetSafe"` query in the help text (after replacing TargetSafe with a real name).  

```
PowerShell Version:5.0
psPAS Branch & Version:2.2.17
CyberArk Version:10.4
```



> Invoke-PASRestMethod : [500] Error mapping types.
> Mapping types:
> IEnumerable`1 -> List`1
> System.Collections.Generic.IEnumerable`1[[CyberArk.PasswordVault.API.Objects.PVPassword, CyberArk.PasswordVault.API, Version=8.0.0.0, Culture=neutral, 
> PublicKeyToken=40be1dbc8718670f]] -> System.Collections.Generic.List`1[[CyberArk.PasswordVault.PASWebServices.Models.AccountModel, CyberArk.PasswordVault.PASWebServices, 
> Version=8.0.0.0, Culture=neutral, PublicKeyToken=40be1dbc8718670f]]
> At C:\Users\***\Documents\WindowsPowerShell\Modules\psPAS\2.2.17\Functions\Accounts\Get-PASAccount.ps1:285 char:13
> + ...   $result = Invoke-PASRestMethod -Uri $URI -Method GET -Headers $sess ...
> +                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>     + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
>     + FullyQualifiedErrorId : CAWS00001E,Invoke-PASRestMethod

**Comment by pspete** [comment]
That is some intriguing  additional info @SphericalMishap - thanks for confirming that this also happens when not using the psPAS module - I am still unable to replicate this issue in my environment.
For your results, is there anyway to figure out what lies at item at # 128?

**Comment by SphericalMishap** [comment]
I've been able to identify at least three specific items that cause the error, but in comparing them to similar objects I haven't found a common factor among them. I'll try to anonymize what I can below while keeping syntax intact. Maybe a second set of eyes will pick up on something. 

**Problem Item #1 - Windows Local Account**
Name: hostname.domain.local-username-random-text-characters
Platform: XXX_Local_Desktop_Admin
Safe: WIN-LOCAL-DSK-ADM
Address: XXX123456.domain.local
Username: ABCD
Logon Account: None
Reconcile Account: XXX_Really_Long_Name_here-username-domain.local

**Successful Item #1 - Windows Local Account**
Name: XXX_WIN_L_Admin_API_Testing-hostname.domain.local-username
Platform: XXX_Win_L_Admin_API_Testing
Safe: TEST-API-SAFE
Address: XXX123456.domain.local (identical to previous example)
Username: EFGH
Logon Account: None
Reconcile Account: XXX_Really_Long_Name_here-diffusername-domain.local



**Problem Item #2 - Unix Root**
Name: Operating System-XXX_Unmanaged_Generic-127.0.0.1-root
Platform: TEST_Unix_SSH
Safe: Safe-Name
Address: 127.0.0.1
Username: root
Logon Account: None
Reconcile Account: XXX_Really_Long_Name_here-username-SSH Key Reconcile Object

**Successful Item #2 - Unix Root**
Name: vmware root
Platform: XXX_Unmanaged_Generic
Safe: VMWare-XX
Address: vmware
Username: root
Logon Account: None
Reconcile Account: None

**Comment by gdv26z** [comment]
Hi
I have the same problem
I wont to get a list of all PASAccounts
$token | Get-PASAccount
I get next Error


My Environment
PowerShell Version:5.1
psPAS  Version:2.3.0
CyberArk Version:10.4

writeErrorStream      : True
Exception             : Microsoft.PowerShell.Commands.WriteErrorException: [500] Error mapping types.
                        
                        Mapping types:
                        IEnumerable`1 -> List`1
                        System.Collections.Generic.IEnumerable`1[[CyberArk.PasswordVault.API.Objects.PVPassword, CyberArk.PasswordVault.API, Version=8.0.0.0, 
                        Culture=neutral, PublicKeyToken=40be1dbc8718670f]] -> 
                        System.Collections.Generic.List`1[[CyberArk.PasswordVault.PASWebServices.Models.AccountModel, CyberArk.PasswordVault.PASWebServices, 
                        Version=8.0.0.0, Culture=neutral, PublicKeyToken=40be1dbc8718670f]]
TargetObject          : 
CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
FullyQualifiedErrorId : CAWS00001E,Invoke-PASRestMethod
ErrorDetails          : 
InvocationInfo        : System.Management.Automation.InvocationInfo
ScriptStackTrace      : at Invoke-PASRestMethod<Process>, C:\Windows\system32\WindowsPowerShell\v1.0\Modules\psPAS\Private\Invoke-PASRestMethod.ps1: line 180
                        at Get-PASAccount<Process>, C:\Windows\system32\WindowsPowerShell\v1.0\Modules\psPAS\Functions\Accounts\Get-PASAccount.ps1: line 285
                        at <ScriptBlock>, <No file>: line 1
PipelineIterationInfo : {0, 1}
PSMessageDetails      :

**Comment by gdv26z** [comment]
Hi
I have the same problem
I wont to get a list of all PASAccounts
$token | Get-PASAccount
I get next Error

My Environment
PowerShell Version:5.1
psPAS Version:2.3.0
CyberArk Version:10.4


`writeErrorStream      : True
Exception             : Microsoft.PowerShell.Commands.WriteErrorException: [500] Error mapping types.
                        
                        Mapping types:
                        IEnumerable`1 -> List`1
                        System.Collections.Generic.IEnumerable`1[[CyberArk.PasswordVault.API.Objects.PVPassword, CyberArk.PasswordVault.API, Version=8.0.0.0, 
                        Culture=neutral, PublicKeyToken=40be1dbc8718670f]] -> 
                        System.Collections.Generic.List`1[[CyberArk.PasswordVault.PASWebServices.Models.AccountModel, CyberArk.PasswordVault.PASWebServices, 
                        Version=8.0.0.0, Culture=neutral, PublicKeyToken=40be1dbc8718670f]]
TargetObject          : 
CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
FullyQualifiedErrorId : CAWS00001E,Invoke-PASRestMethod
ErrorDetails          : 
InvocationInfo        : System.Management.Automation.InvocationInfo
ScriptStackTrace      : at Invoke-PASRestMethod<Process>, C:\Windows\system32\WindowsPowerShell\v1.0\Modules\psPAS\Private\Invoke-PASRestMethod.ps1: line 180
                        at Get-PASAccount<Process>, C:\Windows\system32\WindowsPowerShell\v1.0\Modules\psPAS\Functions\Accounts\Get-PASAccount.ps1: line 285
                        at <ScriptBlock>, <No file>: line 1
PipelineIterationInfo : {0, 1}
PSMessageDetails      :  
`

**Comment by gdv26z** [comment]
Hi
I have the same problem
I want to get a list all CyberArk Password Objects

My Environment
PowerShell Version: 5.1
psPAS Version: 2.3.0
CyberArk Version:10.4

### writeErrorStream    : True
Exception             : Microsoft.PowerShell.Commands.WriteErrorException: [500] Error mapping types.
                        Mapping types:
                        IEnumerable1 -> List1
                        System.Collections.Generic.IEnumerable`1[[CyberArk.PasswordVault.API.Objects.PVPassword, CyberArk.PasswordVault.API, Version=8.0.0.0, 
                        Culture=neutral, PublicKeyToken=40be1dbc8718670f]] -> 
                        System.Collections.Generic.List`1[[CyberArk.PasswordVault.PASWebServices.Models.AccountModel, CyberArk.PasswordVault.PASWebServices, 
                        Version=8.0.0.0, Culture=neutral, PublicKeyToken=40be1dbc8718670f]]
TargetObject          : 
CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
FullyQualifiedErrorId : CAWS00001E,Invoke-PASRestMethod
ErrorDetails          : 
InvocationInfo        : System.Management.Automation.InvocationInfo
ScriptStackTrace      : at Invoke-PASRestMethod<Process>, C:\Windows\system32\WindowsPowerShell\v1.0\Modules\psPAS\Private\Invoke-PASRestMethod.ps1: line 180
                        at Get-PASAccount<Process>, C:\Windows\system32\WindowsPowerShell\v1.0\Modules\psPAS\Functions\Accounts\Get-PASAccount.ps1: line 285
                        at <ScriptBlock>, <No file>: line 1
PipelineIterationInfo : {0, 1}
PSMessageDetails      :

**Comment by Thomas-Kaiser** [comment]
Hi,

i have exactly the same issue. My problematic account is 470.
469 works:
 

```
$res= Invoke-RestMethod -Method GET -Headers $headers -URI ($token.BaseURI+"/PasswordVault/api/Accounts?offset=469&limit=1&filter=safename+eq+*BCN_EMEA_all") -ContentType 'application/json' -TimeoutSec 60
$res.value
```
```
id                        : 12565_12
name                      : Unix-zgzslxrvbre02-root
address                   : zgzslxrvbre02.xxxxxxxxxxxxx 
```

Next account (offset = 470)
```
$res = Invoke-RestMethod -Method GET -Headers $headers -URI ($token.BaseURI+"/PasswordVault/api/Accounts?offset=470&limit=1&filter=safename+eq+*BCN_EMEA_all") -ContentType 'application/json' -TimeoutSec 60
$res.value 
```
```
Invoke-RestMethod : {"ErrorCode":"CAWS00001E","ErrorMessage":"Error mapping types.\r\n\r\nMapping types:\r\nIEnumerable`1 -> 
List`1\r\nSystem.Collections.Generic.IEnumerable`1[[CyberArk.PasswordVault.API.Objects.PVPassword, CyberArk.PasswordVault.API, Version=8.0.0.0, Culture=neutral, 
PublicKeyToken=40be1dbc8718670f]] -> System.Collections.Generic.List`1[[CyberArk.PasswordVault.PASWebServices.Models.AccountModel, 
CyberArk.PasswordVault.PASWebServices, Version=8.0.0.0, Culture=neutral, PublicKeyToken=40be1dbc8718670f]]"} 
```

Did somefound a root cause or any hints what i can do?

**Comment by FIMTooler** [comment]
I've had similar errors from the API when doing a basic search by platformID using the Get Accounts method of the API v10.4.

Using a loop with an the NextLink property (not documented in 10.4), I did a search with a result limit of 1 to find out that it was the 16th account causing the issue.  The results order is same as results in PVWA search, so that's how I located which account was 16th.

I reviewed the CyberArk.WebApplication.log file to find a much longer stack trace that was probably 20 calls long.  About the 4th call from the top had an very revealing error.

System.FormatException: String was not recognized as a valid DateTime.

This pointed me to one of the categories that holds time.  This particular account had 4 (last success verify, last success change, last success reconcile, last fail).  From there I removed one at a time until the above mentioned search loop got past the 16th item.  In my case it was the last success reconcile date causing the issue.  I'm not sure as to why that was the case.

I have a short powershell to convert the time stored in the vault to a readable format and it correctly came out with a valid date time when converting it's value.
